evaluator.go

```diff
@@ -153,6 +171,31 @@ func (state *State) Query(input string) (Result, error) {
 	return evalRangeWithContext(input, state, &context)
 }
 
+// Used for populating state.cachedclusterQueryResult
+// Go over all clusters in state.clusters, look if element is part of cluster definition,
+// if yes, add it name of cluster to cachedCQR[element]
+// as a precaution, we donot want to cache empty results
+func (state *State) populateCachedCQRforSlice(slice []interface{}) {
```

Comment:
If not going to fix `PrimeCache` now, please add a caveat to documentation for it that it doesn't actually do what it says it's going to do